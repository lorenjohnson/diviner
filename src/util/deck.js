import { difference } from 'lodash/fp'
import { random } from 'lodash'
import PropTypes from 'prop-types'

export function getRandomCard (allCards, drawnCards) {
  const remainingCards = difference(allCards, drawnCards)
  if (remainingCards.length === 0) return null
  const drawnCardIndex = random(remainingCards.length - 1)

  return remainingCards[drawnCardIndex]
}

export const DECK_PROP_TYPES = PropTypes.shape({
  backImageURL: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired
})

export const DEFAULT_DECK_BACK_IMAGE = 'https://i.pinimg.com/originals/74/b3/55/74b355023cfe202a3ee3d9c21175f0a3.png'

export { default as DEFAULT_DECK } from 'data/rider-waite-deck.json'
