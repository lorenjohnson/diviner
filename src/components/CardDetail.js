import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { CARD_PROP_TYPES } from 'util/deck'

export default function CardDetail ({ card, className }) {
  return <div className={classNames(className)}>
    <h1>{card.name}</h1>
    <p>{card.desc}</p>
    <h2>Meaning</h2>
    <p>{card.meaningLong}</p>
  </div>
}

CardDetail.propTypes = {
  card: PropTypes.shape(CARD_PROP_TYPES).isRequired,
  className: PropTypes.string
}
