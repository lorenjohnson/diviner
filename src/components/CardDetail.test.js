import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import { SAMPLE_CARD } from 'util/deck'
import CardDetail from './CardDetail'

it('matches last snapshot', () => {
  const component = ReactTestRenderer.create(
    <CardDetail card={SAMPLE_CARD} />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
