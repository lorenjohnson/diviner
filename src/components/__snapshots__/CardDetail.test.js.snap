// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
<div
  className=""
>
  <h1 />
  <p>
    Card 1 description
  </p>
  <h2>
    Meaning
  </h2>
  <p>
    Long very meaningful words.
  </p>
</div>
`;
